core/model/__init__.py

```diff
@@ -334,38 +334,40 @@ def initialize(cls, url, initialize_data=True, initialize_schema=True):
             return engine, engine.connect()
 
         engine = cls.engine(url)
-        if initialize_schema:
-            cls.initialize_schema(engine)
-        connection = engine.connect()
-
-        # Check if the recursive equivalents function exists already.
-        query = (
-            select([literal_column("proname")])
-            .select_from(table("pg_proc"))
-            .where(literal_column("proname") == "fn_recursive_equivalents")
-        )
-        result = connection.execute(query)
-        result = list(result)
-
-        # If it doesn't, create it.
-        if not result and initialize_data:
-            resource_file = os.path.join(
-                cls.resource_directory(), cls.RECURSIVE_EQUIVALENTS_FUNCTION
+        with engine.connect() as connection:
+            tx = connection.begin()
+            connection.execute("LOCK TABLE pg_proc IN ACCESS EXCLUSIVE MODE;")
```

Comment:
Assuming that a CM doesn't have "root" access to the DB server, will it be able to successfully get a lock on this table?

Reply:
I'm not sure.  If there were a problem, wouldn't docker-compose or the other docker tests have failed?

Reply:
I'll try it on one of the dev rds databases.

Reply:
Good catch @tdilauro :  this doesn't work on RDS - you need to be root.